plugins/kubernetes-backend/src/kubernetes-auth-translator/KubernetesAuthTranslatorGenerator.test.ts

```diff
@@ -14,46 +14,78 @@
  * limitations under the License.
  */
 
-import { KubernetesAuthTranslator } from './types';
-import { GoogleKubernetesAuthTranslator } from './GoogleKubernetesAuthTranslator';
 import { KubernetesAuthTranslatorGenerator } from './KubernetesAuthTranslatorGenerator';
-import { NoopKubernetesAuthTranslator } from './NoopKubernetesAuthTranslator';
-import { AwsIamKubernetesAuthTranslator } from './AwsIamKubernetesAuthTranslator';
-import { OidcKubernetesAuthTranslator } from './OidcKubernetesAuthTranslator';
-import { getVoidLogger } from '@backstage/backend-common';
+import { ClusterDetails } from '../types';
+import { KubernetesRequestAuth } from '@backstage/plugin-kubernetes-common';
 
-const logger = getVoidLogger();
+const mockTranslator = jest.fn();
 
-describe('getKubernetesAuthTranslatorInstance', () => {
-  const sut = KubernetesAuthTranslatorGenerator;
+const authTranslatorMap = {
+  google: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  aws: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  azure: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  serviceAccount: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  googleServiceAccount: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  oidc: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+  localKubectlProxy: {
+    decorateClusterDetailsWithAuth: mockTranslator,
+  },
+};
 
-  it('can return an auth translator for google auth', () => {
-    const authTranslator: KubernetesAuthTranslator =
-      sut.getKubernetesAuthTranslatorInstance('google', { logger });
-    expect(authTranslator instanceof GoogleKubernetesAuthTranslator).toBe(true);
-  });
+describe('decorateClusterDetailsWithAuth', () => {
+  let authTranslator: KubernetesAuthTranslatorGenerator;
+  const authObject: KubernetesRequestAuth = {};
 
-  it('can return an auth translator for aws auth', () => {
-    const authTranslator: KubernetesAuthTranslator =
-      sut.getKubernetesAuthTranslatorInstance('aws', { logger });
-    expect(authTranslator instanceof AwsIamKubernetesAuthTranslator).toBe(true);
+  beforeEach(() => {
+    mockTranslator.mockClear();
+    authTranslator = new KubernetesAuthTranslatorGenerator({
+      authTranslatorMap,
```

Comment:
This is the toned-down `authTranslatorMap`, just complex enough to exercise the generator fully
```suggestion
      authTranslatorMap: { google: mockTranslator },
```

Reply:
Done